core/src/main/java/de/danoeh/antennapod/core/sync/SyncService.java

```diff
@@ -292,36 +202,25 @@ public static long getLastSyncAttempt(Context context) {
     private List<String> getQueuedRemovedFeeds() {
         ArrayList<String> actions = new ArrayList<>();
         try {
-            SharedPreferences prefs = getApplicationContext().getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE);
-            String json = prefs.getString(PREF_QUEUED_FEEDS_REMOVED, "[]");
-            JSONArray queue = new JSONArray(json);
-            for (int i = 0; i < queue.length(); i++) {
-                actions.add(queue.getString(i));
-            }
+            actions = SynchronizationQueue.getQueuedRemovedFeeds();
         } catch (JSONException e) {
             e.printStackTrace();
         }
         return actions;
     }
 
     private List<String> getQueuedAddedFeeds() {
-        ArrayList<String> actions = new ArrayList<>();
+        ArrayList<String> addedFeedUrls = new ArrayList<>();
         try {
-            SharedPreferences prefs = getApplicationContext().getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE);
-            String json = prefs.getString(PREF_QUEUED_FEEDS_ADDED, "[]");
-            JSONArray queue = new JSONArray(json);
-            for (int i = 0; i < queue.length(); i++) {
-                actions.add(queue.getString(i));
-            }
+            addedFeedUrls = SynchronizationQueue.getQueuedAddedFeeds();
```

Comment:
How about handling try/catch inside `SynchronizationQueue` and removing this method here completely? This could make `SyncService` a nice and small class :)